### PR TITLE
Fix issue in recovery V2 /confirm API for EXTERNAL channel

### PR DESCRIPTION
**Pull request by Rashmini**
### Proposed changes in this pull request

$subject

### Related issue

- https://github.com/wso2/product-is/issues/16884

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/password/PasswordRecoveryManagerImpl.java [hunk 1]
```diff
@@ -910,17 +910,27 @@ private PasswordResetCodeDTO validateConfirmationCode(UserAccountRecoveryManager
...
+                }
+                throw ex;
             }
-            throw e;
         }
-        if (!StringUtils.equals(userRecoveryData.getRemainingSetIds(),
-                NotificationChannels.EMAIL_CHANNEL.getChannelType())) {
+        if (!(StringUtils.equals(userRecoveryData.getRemainingSetIds(),NotificationChannels.EMAIL_CHANNEL.
+                getChannelType()) || StringUtils.equals(userRecoveryData.getRemainingSetIds(),NotificationChannels.
+                EXTERNAL_CHANNEL.getChannelType()))) {
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/password/PasswordRecoveryManagerImpl.java:933]:
```suggestion
        if (!(StringUtils.equals(userRecoveryData.getRemainingSetIds(), NotificationChannels.EMAIL_CHANNEL.
                getChannelType()) || StringUtils.equals(userRecoveryData.getRemainingSetIds(), NotificationChannels.
                EXTERNAL_CHANNEL.getChannelType()))) {
```
formatting issue

Better to use <Constant>.equals(<variable>) format

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java [hunk 2]
```diff
@@ -803,16 +803,27 @@ private UserRecoveryData validateUserRecoveryDataFromCode(String code, String co
...
+                }
+                throw ex;
             }
-            throw e;
         }
-        if (!StringUtils.equals(userRecoveryData.getRemainingSetIds(),
-                NotificationChannels.EMAIL_CHANNEL.getChannelType())) {
+        if (!(StringUtils.equals(userRecoveryData.getRemainingSetIds(),NotificationChannels.EMAIL_CHANNEL.
+                getChannelType()) || StringUtils.equals(userRecoveryData.getRemainingSetIds(),NotificationChannels.
+                EXTERNAL_CHANNEL.getChannelType()))) {
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java:826]: same comment as https://github.com/wso2-extensions/identity-governance/pull/770/files#r1350463622

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java [hunk 3]
```diff
@@ -1204,8 +1204,9 @@ public static String generateSecretKey(String channel, String tenantDomain, Stri
      */
     public static String concatRecoveryFlowIdWithSecretKey(String recoveryFlowId, String notificationChannel,
                                                            String secretKey) {
-        if (recoveryFlowId != null && StringUtils.equals(notificationChannel,
-                NotificationChannels.EMAIL_CHANNEL.getChannelType())) {
+        if (recoveryFlowId != null && (StringUtils.equals(notificationChannel, NotificationChannels.EMAIL_CHANNEL.
+                getChannelType()) || StringUtils.equals(notificationChannel, NotificationChannels.EXTERNAL_CHANNEL.
+                getChannelType()))) {
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java:1209]: use `<constant>.equals.(<variable>)` format

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/password/PasswordRecoveryManagerImpl.java (hunk 1 above)
Comment (Rashmini) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/password/PasswordRecoveryManagerImpl.java]: Fixed with 6e6aa3151db682c8b69adc0ed7e8f068a8e713e2

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java (hunk 2 above)
Comment (Rashmini) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Fixed with 6e6aa3151db682c8b69adc0ed7e8f068a8e713e2

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java (hunk 3 above)
Comment (Rashmini) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java]: Fixed with 6e6aa3151db682c8b69adc0ed7e8f068a8e713e2

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**